cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ReplicationThrottleHelper.java

```diff
@@ -390,4 +396,64 @@ static boolean configsEqual(Config configs, Map<String, String> expectedValues)
     }
     return true;
   }
+
+  private enum LogConfig {
+    LEADER_REPLICATION_THROTTLED_REPLICAS_CONFIG,
+    FOLLOWER_REPLICATION_THROTTLED_REPLICAS_CONFIG,
+  }
+
+  /**
+   * Starting with Kafka 3.5.0, the location of the "LogConfig" class is "org.apache.kafka.storage.internals.log.LogConfig"
+   * and provides new constant fields in place of some methods of the original class.
+   *
+   *   - LogConfig class in Kafka 3.5+: org.apache.kafka.storage.internals.log.LogConfig
+   *   - LogConfig class in Kafka 3.4-: kafka.log.LogConfig
+   **
+   * The older LogConfig class does not work with the newer versions of Kafka. Therefore, if the new class exists, we use it and if
+   * it doesn't exist we will fall back on the older one.
+   *
+   * Once CC supports only 3.5.0 and newer, we can clean this up and use the LogConfig class from
+   * `org.apache.kafka.storage.internals.log.LogConfig`all the time.
+   * @param config LogConfig config name
+   * @return LogConfig config name in the format of a Kafka configuration property
+   */
+  private static String getLogConfig(LogConfig config) {
+    Class<?> logConfigClass;
+
+    try {
+      // First we try to get the LogConfig class for Kafka 3.5+
+      logConfigClass = Class.forName(LOG_CONFIG_IN_KAFKA_3_5_AND_LATER);
+      LOG.info("Found class {} for Kafka 3.5 and newer.", LOG_CONFIG_IN_KAFKA_3_5_AND_LATER);
+      try {
+        Field field = logConfigClass.getDeclaredField(config.toString());
+        return field.get(null).toString();
+      } catch (NoSuchFieldException | IllegalAccessException e) {
```

Comment:
Thanks for the change! I think the error in L431 and error in L434 are likely always atomic. If we are confident enough about it, I think we can just catch one Exception, and log info with something like "Failed to read XXX config from LogConfig class since we are probably on kafka 3.4 or older: + e (the exception string)". That will further simplify the code. Same with L452 and L456. What do you think?

Reply:
@CCisGG yes that makes sense, just updated the code!